fem/pgridfunc.cpp

```diff
@@ -242,18 +242,19 @@ void ParGridFunction::ExchangeFaceNbrData()
       d_send_data[i] = d_data[d_send_ldof[i]];
    });
 
-   auto h_send_data = send_data.HostRead();
-   auto h_face_nbr_data = face_nbr_data.HostWrite();
+   bool mpi_gpu_aware = Device::GetGPUAwareMPI();
+   auto send_data_ptr = mpi_gpu_aware ? send_data.Read() : send_data.HostRead();
+   auto face_nbr_data_ptr = mpi_gpu_aware ? face_nbr_data.Write() : face_nbr_data.HostWrite();
    for (int fn = 0; fn < num_face_nbrs; fn++)
```

Comment:
Nice! Can you let me know this improves performance? I use this function quite a bit.

Reply:
@camierjs Do you have any information on this?

Reply:
I'm using it locally, but haven't re-tested it on some large run.